experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -489,9 +604,51 @@ function reynolds_operator(X::RepresentationLinearlyReductiveGroup, elem::MPolyR
     return reverse_map(f)
 end
 
+@doc raw"""
+    reynolds_operator(RG::RedGrpInvRing, f::MPolyRingElem)
+
+Return the image of `f` under the Reynolds operator corresponding to `RG`.
+
+# Examples
+```jldoctest
+julia> G = linearly_reductive_group(:SL, 3, QQ);
+
+julia> r = representation_on_forms(G, 3);
+
+julia> S, x = graded_polynomial_ring(QQ, "x" => 1:10);
+
+julia> RG = invariant_ring(S, r);
+
+julia> 75*reynolds_operator(RG, x[5]^4)
+x[1]*x[4]*x[8]*x[10] - x[1]*x[4]*x[9]^2 - x[1]*x[5]*x[7]*x[10] + x[1]*x[5]*x[8]*x[9] + x[1]*x[6]*x[7]*x[9] - x[1]*x[6]*x[8]^2 - x[2]^2*x[8]*x[10] + x[2]^2*x[9]^2 + x[2]*x[3]*x[7]*x[10] - x[2]*x[3]*x[8]*x[9] + x[2]*x[4]*x[5]*x[10] - x[2]*x[4]*x[6]*x[9] - 2*x[2]*x[5]^2*x[9] + 3*x[2]*x[5]*x[6]*x[8] - x[2]*x[6]^2*x[7] - x[3]^2*x[7]*x[9] + x[3]^2*x[8]^2 - x[3]*x[4]^2*x[10] + 3*x[3]*x[4]*x[5]*x[9] - x[3]*x[4]*x[6]*x[8] - 2*x[3]*x[5]^2*x[8] + x[3]*x[5]*x[6]*x[7] + x[4]^2*x[6]^2 - 2*x[4]*x[5]^2*x[6] + x[5]^4
+```
+"""
 function reynolds_operator(R::RedGrpInvRing, elem::MPolyRingElem)
     X = R.representation
     return reynolds_operator(X, elem)
 end
 
 include("TorusInvariantsFast.jl")
+
+
+#####################Invariant rings as affine algebras
+
+@doc raw"""
+    affine_algebra(RG::RedGrpInvRing)
+
+Return the invariant ring `RG` as an affine algebra (this amounts to compute the algebra syzygies among the fundamental invariants of `RG`).
+
+In addition, if `A` is this algebra, and `R` is the polynomial ring of which `RG` is a subalgebra,
+return the inclusion homomorphism  `A` $\hookrightarrow$ `R` whose image is `RG`.
+"""
+function affine_algebra(R::RedGrpInvRing)
+   V = fundamental_invariants(R)
+   s = length(V)
+   S,t = polynomial_ring(field(group(representation(R))), "t"=>1:s)
```

Comment:
Assuming the fundamental invariants are homogeneous, we could/should grade this polynomial ring appropriately.
Also, the presentation as affine algebra should be cached in the invariant ring.

This can be filed under "future improvements".

Reply:
@joschmitt As far as I can see, the ring is NOT graded for finite groups. Or?

Reply:
It is graded:
```
julia> RG = invariant_ring(matrix_group(QQ[-1 0 ; 0 -1]));

julia> A, AtoR = affine_algebra(RG);

julia> A
Quotient
  of multivariate polynomial ring in 3 variables over QQ graded by
    y1 -> [2]
    y2 -> [2]
    y3 -> [2]
  by ideal (-y1*y3 + y2^2)
```